checkpoint-answers.txt

```diff
@@ -0,0 +1,46 @@
+Checkpoint Answers
+
+1) What's a RubyGem and why would you use one?
+
+Answer) A gem is essentially a library of code that can be added to and reused in projects.  This allows you to build functionality or extend existing functionality.  This also helps to make code in projects more concise and readable be hiding the underlying, more complex code.
+
+
+2) What's the difference between lazy and eager loading?
+
+Answer) The basic difference is whether something is loaded or compiled at the time of creation.  Creating a stored procedure that will connect to a database and be used to return information at a later time; this would be an example of lazy loading.  Creating an object that is connected to a database so that it is ready to be used now or at a later time to retrieve information; this would be eager loading.  You would want to use eager loading if you know you will be using the data immediately, otherwise, lazy loading uses less resources.
+
+
+3) What's the difference betwee the CREATE TABLE and INSERT INTO SQL statements.  
+
+Answer) CREATE TABLE is used to create a well-defined table to be used.  The INSERT INTO, will create a table dynamically via a SELECT query.  This statement will create a table defined by the data return by the results of the SELECT statement used to create it.  This often means you cannot guarantee the length of text VARCHAR fields or the type of numeric fields.
```

Comment:
`INSERT INTO` does not work quite like you suggest. It creates records and adds them to an existing table. If the table does not exist, then the insert will fail.